metrics/graphite/graphite.go

```diff
@@ -0,0 +1,295 @@
+// Package graphite implements a graphite backend for package metrics.
+//
+// The current implementation ignores fields.
+package graphite
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"math"
+	"net"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/go-kit/kit/metrics"
+)
+
+// Emitter will keep track of all metrics and, once started,
+// will emit the metrics via the Flush method to the given address.
+type Emitter interface {
+	NewCounter(string) metrics.Counter
+	NewHistogram(string, int64, int64, int, ...int) metrics.Histogram
+	NewTimeHistogram(string, time.Duration, int64, int64, int, ...int) metrics.TimeHistogram
+	NewGauge(string) metrics.Gauge
+
+	Start(time.Duration)
+	Flush() error
+}
+
+type emitter struct {
+	addr   *net.TCPAddr
+	prefix string
+
+	metricMu   *sync.Mutex
+	counters   []*counter
+	histograms []*windowedHistogram
+	gauges     []*gauge
+}
+
+// NewEmitter will return an Emitter that will prefix all
+// metrics names with the given prefix. Once started, it will attempt to create
+// a TCP connection with the given address and periodically post metrics to the
+// connection in a Graphite-compatible format.
+func NewEmitter(addr *net.TCPAddr, prefix string) Emitter {
+	e := &emitter{
+		addr, prefix, &sync.Mutex{},
+		[]*counter{}, []*windowedHistogram{}, []*gauge{},
+	}
+
+	return e
```

Comment:
If you go with the above suggestions, and leverage the One Neat Trick that the zero value of a slice type can be appended-to, then this constructor can reduce to

```go
return &emitter{
    addr:    addr,
    prefix: prefix,
}
```

(Always give key prefixes to struct members during construction...)